.github/workflows/static-checks.yml

```diff
@@ -0,0 +1,190 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Static Checks CI
+on:
+  pull_request:
```

Comment:
I think we also need this GitHub Action to run on each commit on master and release branch. 

e.g.

```yaml
on:
  push:
    branches: [ master ]
```

Reply:
it's possible that Infra needs to enable actions for this repo?

Reply:
Seems like it. This runs on a PR raised in the fork repo.

Reply:
https://the-asf.slack.com/archives/CBX4TSBQ8/p1668410098261039 - We need to merge the first PR before it can get triggered.